src/tallies/filter_time.cpp

```diff
@@ -0,0 +1,97 @@
+#include "openmc/tallies/filter_time.h"
+
+#include <algorithm> // for min, max
+
+#include <fmt/core.h>
+
+#include "openmc/search.h"
+#include "openmc/xml_interface.h"
+
+namespace openmc {
+
+//==============================================================================
+// TimeFilter implementation
+//==============================================================================
+
+void TimeFilter::from_xml(pugi::xml_node node)
+{
+  auto bins = get_node_array<double>(node, "bins");
+  this->set_bins(bins);
+}
+
+void TimeFilter::set_bins(gsl::span<const double> bins)
+{
+  // Clear existing bins
+  bins_.clear();
+  bins_.reserve(bins.size());
+
+  // Copy bins, ensuring they are valid
+  for (gsl::index i = 0; i < bins.size(); ++i) {
```

Comment:
Idea from @gridley to use `std::is_sorted`. From mesh.cpp:

`if (not std::is_sorted(bins.begin(), bins.end(), std::less_equal<double>()))`

Reply:
Addressed in 7b43bb97a5c51a5f8a30f1554fac966442795854